godot-core/src/task/async_runtime.rs

```diff
@@ -193,29 +807,68 @@ thread_local! {
     static ASYNC_RUNTIME: RefCell<Option<AsyncRuntime>> = RefCell::new(Some(AsyncRuntime::new()));
 }
 
+/// Simplified lifecycle management for godot engine integration
+///
+/// Note: The original lifecycle module contained extensive monitoring and integration
+/// features that were designed for:
+/// - RuntimeLifecycleState enum: Designed for state tracking during complex initialization
+/// - get_runtime_state/initialize_runtime: Designed for explicit lifecycle management
+/// - on_frame_update/health_check: Designed for runtime monitoring and diagnostics
+/// - engine_integration submodule: Designed for hooking into Godot's lifecycle events
+///
+/// These were removed because the current implementation uses:
+/// - Lazy initialization (runtime created on first use)
+/// - Simple cleanup on engine shutdown
+/// - No need for complex state tracking or health monitoring
+///
+/// Only the essential cleanup function remains.
+pub mod lifecycle {
+    use super::*;
+
+    /// Begin shutdown of the async runtime
+    ///
+    /// Returns the number of tasks that were canceled during shutdown
+    pub fn begin_shutdown() -> usize {
+        ASYNC_RUNTIME.with(|runtime| {
+            if let Some(mut rt) = runtime.borrow_mut().take() {
+                let task_count = rt.task_storage.get_active_task_count();
+
+                // Note: task_count tasks were canceled during shutdown
+
+                // Clear all components
+                rt.clear_all();
+
+                // Drop the runtime to free resources
+                drop(rt);
+
+                task_count
+            } else {
+                0
+            }
+        })
+    }
+}
+
 /// Will be called during engine shutdown.
 ///
 /// We have to drop all the remaining Futures during engine shutdown. This avoids them being dropped at process termination where they would
 /// try to access engine resources, which leads to SEGFAULTs.
 pub(crate) fn cleanup() {
-    ASYNC_RUNTIME.set(None);
+    let _canceled_tasks = lifecycle::begin_shutdown();
+    // Note: _canceled_tasks tasks were canceled during engine shutdown
 }
 
 #[cfg(feature = "trace")]
 pub fn has_godot_task_panicked(task_handle: TaskHandle) -> bool {
-    ASYNC_RUNTIME.with_runtime(|rt| rt.panicked_tasks.contains(&task_handle.id))
+    ASYNC_RUNTIME.with_runtime(|rt| rt.has_task_panicked(task_handle.id as u64))
 }
 
 /// The current state of a future inside the async runtime.
 enum FutureSlotState<T> {
-    /// Slot is currently empty.
-    Empty,
     /// Slot was previously occupied but the future has been canceled or the slot reused.
     Gone,
     /// Slot contains a pending future.
     Pending(T),
-    /// Slot contains a future which is currently being polled.
-    Polling,
 }
```

Comment:
If we keep supporting the Godot runtime, we can probably not just remove these 2 states, can we?